client/packages/react/src/next-ssr/HydrationStreamProvider.tsx

```diff
@@ -0,0 +1,193 @@
+'use client';
```

Comment:
It may make sense to add a top-level comment, (a) describing what a HydrationStream is, (b) describing what this is meant to do.